packages/main/src/components/ActionSheet/index.tsx

```diff
@@ -53,14 +103,24 @@ const ActionSheet: FC<ActionSheetPropTypes> = forwardRef(
       onAfterClose,
       onAfterOpen,
       onBeforeClose,
-      onBeforeOpen
+      onBeforeOpen,
+      showCancelButton,
+      alwaysShowHeader
     } = props;
-
+    const i18nBundle = useI18nBundle('@ui5/webcomponents-react');
     const classes = useStyles();
-
     const actionSheetClasses = StyleClassHelper.of(classes.actionSheet).putIfPresent(className);
-
     const popoverRef: RefObject<Ui5ResponsivePopoverDomRef> = useConsolidatedRef(ref);
+    const actionBtnsRef = useRef(null);
+    const [focusedItem, setFocusedItem] = useReducer((_, action) => {
+      return parseInt(action.target.dataset.actionBtnIndex);
+    }, 0);
+    const childrenArrayLength = Children.toArray(children).length;
```

Comment:
Might be an issue with conditional rendered buttons as childrens:
```suggestion
    const childrenToRender = Children.toArray(children).filter(Boolean);
    const childrenArrayLength = childrenToRender.length;
```